specifyweb/frontend/js_src/lib/components/Formatters/Formatter.tsx

```diff
@@ -145,51 +147,142 @@ function Definitions({
+          key={index}
+        >
           {hasCondition && (
             <Label.Block>
-              {resourcesText.conditionFieldValue()}
-              <Input.Text
-                isReadOnly={isReadOnly}
-                value={value ?? ''}
-                onValueChange={(value): void =>
-                  handleChanged(
-                    {
-                      value: value.length === 0 ? undefined : value,
-                      fields,
-                    },
-                    index
-                  )
-                }
-              />
-              <span>
-                {index === 0
-                  ? resourcesText.elseConditionDescription()
-                  : resourcesText.conditionDescription()}
-              </span>
+              {showConditionalField[index]
+                ? resourcesText.conditionFieldValue()
+                : null}
+              <div className="flex items-center gap-2">
+                <Input.Text
+                  className="h-full"
+                  isReadOnly={isReadOnly}
+                  value={value ?? ''}
+                  onValueChange={(value): void =>
+                    handleChanged(
+                      {
+                        value: value.length === 0 ? undefined : value,
+                        fields,
+                      },
+                      index
+                    )
+                  }
+                />
+                {trimmedFields.length > 0 && showConditionalField[index] ? (
+                  <Button.Danger
+                    className="w-[40%]"
+                    onClick={(): void =>
+                      handleChange(
+                        removeItem(formatter.definition.fields, index)
+                      )
+                    }
+                  >
+                    {resourcesText.deleteDefinition()}
+                  </Button.Danger>
+                ) : null}
+              </div>
+              {showConditionalField[index] ? (
+                <span>
+                  {index === 0
+                    ? resourcesText.elseConditionDescription()
+                    : resourcesText.conditionDescription()}
+                </span>
+              ) : null}
             </Label.Block>
           )}
-          <Fields
-            fields={[
-              fields,
-              (fields): void => handleChanged({ value, fields }, index),
-            ]}
-            table={table}
-            onDelete={
-              trimmedFields.length < 2
-                ? undefined
-                : (): void =>
-                    handleChange(removeItem(formatter.definition.fields, index))
-            }
-          />
+          {showConditionalField[index] ||
+          !hasCondition ? null : fields.length === 0 ? (
+            <Button.Small
+              onClick={(): void => {
+                const newConditionalField = Array.from(showConditionalField);
+                newConditionalField[index] = !newConditionalField[index];
```

Comment:
fyi, could have used `replaceItem` util here

Reply:
when using replaceItem I have a typescript error `Argument of type 'RA<boolean>' is not assignable to parameter of type 'SetStateAction<boolean[]>'.
  The type 'readonly boolean[]' is 'readonly' and cannot be assigned to the mutable type 'boolean[]'.`

Reply:
Same thing for the `onRemoveField() `

Reply:
see https://github.com/specify/specify7/pull/4381#discussion_r1449555484